server/mscalendar/availability.go

```diff
@@ -21,78 +22,101 @@ const (
 	StatusSyncJobInterval           = 5 * time.Minute
 	upcomingEventNotificationTime   = 10 * time.Minute
 	upcomingEventNotificationWindow = (StatusSyncJobInterval * 11) / 10 // 110% of the interval
+	logTruncateMsg                  = "We've truncated the logs due to too many messages"
+	logTruncateLimit                = 5
 )
 
+type StatusSyncJobSummary struct {
+	NumberOfUsersFailedStatusChanged int
+	NumberOfUsersStatusChanged       int
+	NumberOfUsersProcessed           int
+}
+
 type Availability interface {
 	GetCalendarViews(users []*store.User) ([]*remote.ViewCalendarResponse, error)
-	Sync(mattermostUserID string) (string, error)
-	SyncAll() (string, error)
+	Sync(mattermostUserID string) (string, *StatusSyncJobSummary, error)
+	SyncAll() (string, *StatusSyncJobSummary, error)
 }
 
-func (m *mscalendar) Sync(mattermostUserID string) (string, error) {
+func (m *mscalendar) Sync(mattermostUserID string) (string, *StatusSyncJobSummary, error) {
 	user, err := m.Store.LoadUserFromIndex(mattermostUserID)
 	if err != nil {
-		return "", err
+		return "", nil, err
 	}
 
 	return m.syncUsers(store.UserIndex{user})
 }
 
-func (m *mscalendar) SyncAll() (string, error) {
+func (m *mscalendar) SyncAll() (string, *StatusSyncJobSummary, error) {
 	err := m.Filter(withSuperuserClient)
 	if err != nil {
-		return "", err
+		return "", nil, errors.Wrap(err, "not able to filter the super user client.")
 	}
 
 	userIndex, err := m.Store.LoadUserIndex()
 	if err != nil {
 		if err.Error() == "not found" {
-			return "No users found in user index", nil
+			return "No users found in user index", nil, nil
 		}
-		return "", err
+		return "", nil, errors.Wrap(err, "not able to load the users from user index.")
 	}
 
 	return m.syncUsers(userIndex)
 }
 
-func (m *mscalendar) syncUsers(userIndex store.UserIndex) (string, error) {
+func (m *mscalendar) syncUsers(userIndex store.UserIndex) (string, *StatusSyncJobSummary, error) {
+	syncJobSummary := &StatusSyncJobSummary{}
 	if len(userIndex) == 0 {
-		return "No connected users found", nil
+		return "No connected users found", syncJobSummary, nil
 	}
+	syncJobSummary.NumberOfUsersProcessed = len(userIndex)
 
+	numberOfLogs := 0
 	users := []*store.User{}
 	for _, u := range userIndex {
 		// TODO fetch users from kvstore in batches, and process in batches instead of all at once
 		user, err := m.Store.LoadUser(u.MattermostUserID)
 		if err != nil {
-			return "", err
+			if numberOfLogs < logTruncateLimit {
+				m.Logger.Warnf("Not able to load user %s from user index. err=%v", u.MattermostUserID, err)
+			} else if numberOfLogs == logTruncateLimit {
+				m.Logger.Warnf(logTruncateMsg)
+			}
```

Comment:
What is the value of posing a truncated message instead of the "real" one?

Reply:
@hanzei Please read this comment https://github.com/mattermost/mattermost-plugin-mscalendar/issues/244#issuecomment-1402281372.

Reply:
@mickmister I don't understand your point in https://github.com/mattermost/mattermost-plugin-mscalendar/issues/244#issuecomment-1402281372. If the goal is not to flood the logs, why are messages logged that only contain the information that the logs got trunkated? Isn't the number of log messages still the same?

Reply:
@hanzei No the number of logs will not be the same. We will log the first 4 logs coming from a particular function/code snippet, then for the 5th one we will log the Truncate and after that, we will not be logging anything. So the max number of logs will be 5 in any case. If it is more than that we are truncating means not logging anything for a particular function.